chain/network/src/peer_manager/network_state/mod.rs

```diff
@@ -386,72 +475,150 @@ impl NetworkState {
+                        None
+                    }
+                })
+            })
+            .await;
+            this.routing_table_view.add_local_edges(&edges);
+
+            let mut new_edges = match this
                 .routing_table_addr
-                .send(routing::actor::Message::AddVerifiedEdges { edges }.with_span_context())
+                .send(
+                    routing::actor::Message::AddVerifiedEdges { edges: edges.clone() }
+                        .with_span_context(),
+                )
                 .await
             {
-                Ok(routing::actor::Response::AddVerifiedEdgesResponse(mut edges)) => {
-                    this.config.event_sink.push(Event::EdgesVerified(edges.clone()));
-                    // Don't send tombstones during the initial time.
-                    // Most of the network is created during this time, which results
-                    // in us sending a lot of tombstones to peers.
-                    // Later, the amount of new edges is a lot smaller.
-                    if let Some(skip_tombstones_duration) = this.config.skip_tombstones {
-                        if clock.now() < this.start_time + skip_tombstones_duration {
-                            edges.retain(|e| e.edge_type() == EdgeState::Active);
-                            metrics::EDGE_TOMBSTONE_SENDING_SKIPPED.inc();
-                        }
-                    }
-                    this.tier2.broadcast_message(Arc::new(PeerMessage::SyncRoutingTable(
-                        RoutingTableUpdate::from_edges(edges),
-                    )));
+                Ok(routing::actor::Response::AddVerifiedEdges(new_edges)) => new_edges,
+                Ok(_) => panic!("expected AddVerifiedEdges"),
+                Err(err) => {
+                    tracing::error!(target:"network","routing::actor::Actor closed: {err}");
+                    return Ok(());
+                }
+            };
+            this.config.event_sink.push(Event::EdgesVerified(edges.clone()));
+            // Don't send tombstones during the initial time.
+            // Most of the network is created during this time, which results
+            // in us sending a lot of tombstones to peers.
+            // Later, the amount of new edges is a lot smaller.
+            if let Some(skip_tombstones_duration) = this.config.skip_tombstones {
+                if clock.now() < this.created_at + skip_tombstones_duration {
+                    new_edges.retain(|edge| edge.edge_type() == EdgeState::Active);
+                    metrics::EDGE_TOMBSTONE_SENDING_SKIPPED.inc();
                 }
-                _ => tracing::error!(target: "network", "expected AddVerifiedEdgesResponse"),
             }
-        });
+            // Broadcast new edges to all other peers.
+            this.broadcast_routing_table_update(Arc::new(RoutingTableUpdate::from_edges(
+                new_edges,
+            )))
+            .await;
```

Comment:
Hm, this `.await` is suspicious. This will wait until we've send out all updates, and that would wait for `mux` on every connection. So, essentially, the slowest peer blocks all. Is this fine? 

Intuitively, actualy broadcasting should happen in background.

Reply:
With structured concurrency by default you are not supposed to "spawn stuff in the background", but rather return only once all the work is done. On the other hand, with the current implementation it doesn't really matter, because we block on the rate limiter, but on on the network transmission (which actually does happen in the background). IMO broadcast should be blocking and it is up to the caller to decide whether to await for it, or to run it in parallel to sth else.

Reply:
but you are also right, that banning the peer immediately rather than waiting for broadcast might be a good thing to do. For simplicity, I'll make the broadcast spawn stuff in the background for now then.